homeassistant/components/zha/config_flow.py

```diff
@@ -53,48 +183,261 @@ async def async_step_user(self, user_input=None):
+                    value = 115200
+
+                schema[param] = value
+
+        return self.async_show_form(
+            step_id="manual_port_config",
+            data_schema=vol.Schema(schema),
+            errors=errors,
+        )
+
+    async def async_step_choose_formation_strategy(self, user_input=None):
+        """Choose how to deal with the current radio's settings."""
+
+        if user_input is not None:
+            strategy = user_input[FORMATION_STRATEGY]
+
+            if strategy == FORMATION_FORM_NEW_NETWORK:
+                async with self._connect_zigpy_app() as app:
+                    await app.form_network()
+            elif strategy == FORMATION_REUSE_SETTINGS:
+                pass
+            elif strategy == FORMATION_RESTORE_MANUAL_BACKUP:
+                return await self.async_step_upload_manual_backup()
+            elif strategy == FORMATION_RESTORE_AUTOMATIC_BACKUP:
+                return await self.async_step_restore_automatic_backup()
+
+            return await self._async_create_radio_entity()
+
+        async with self._connect_zigpy_app() as app:
+            # Check if the stick has any settings and load them
+            try:
+                await app.load_network_info()
+            except NetworkNotFormed:
+                pass
+            else:
+                # Load the current info while we're connected, to save time
+                self._backups = app.backups.backups.copy()
+                self._current_settings = zigpy.backups.NetworkBackup(
+                    network_info=app.state.network_info,
+                    node_info=app.state.node_info,
+                )
+
+        strategies = []
+
+        # Check if we have any automatic backups *and* if the backups differ from
+        # the current radio settings (since restoring would be redundant)
+        if self._backups and any(
+            not backup.is_compatible_with(self._current_settings)
+            for backup in self._backups
+        ):
+            strategies.append(FORMATION_RESTORE_AUTOMATIC_BACKUP)
+
+        if self._current_settings is not None:
+            strategies.append(FORMATION_REUSE_SETTINGS)
+
+        strategies.append(FORMATION_RESTORE_MANUAL_BACKUP)
+        strategies.append(FORMATION_FORM_NEW_NETWORK)
+
+        # Pick a default. Forming a new network will always be an option but we would
+        # prefer to suggest something non-destructive if it's available.
+        suggested_strategy = [
+            strategy
+            for strategy in strategies
+            if strategy
+            in (
+                FORMATION_RESTORE_AUTOMATIC_BACKUP,
+                FORMATION_REUSE_SETTINGS,
+                FORMATION_FORM_NEW_NETWORK,
+            )
+        ][0]
+
+        schema = {
+            vol.Required(FORMATION_STRATEGY, default=suggested_strategy): vol.In(
+                strategies
+            ),
+        }
+
+        return self.async_show_form(
```

Comment:
Instead of a form, can we use a "menu" ? [Example](https://github.com/home-assistant/core/blob/dev/homeassistant/components/awair/config_flow.py#L82). Will allow translations too.

![image](https://user-images.githubusercontent.com/1444314/185961952-d0b83d5f-1211-432d-ba3b-4858875147b8.png)

Reply:
Oh nice, I wasn't aware there was a component for this. Should we use it as well for selecting a serial port (the first dialog shown when you try to add ZHA)?

Reply:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/32534428/185989780-b306ddab-78fc-4977-9d57-b3326a50dcc7.png">

It looks much nicer. However, there is no longer a default: it's only suggested by the order of the menu options.  Is something like a `(suggested)` suffix good? It doesn't appear there is a way to hint to the frontend to make on `<mwc-list-item>` more prominent than the others.